app/src/main/java/org/mifos/mobile/ui/recent_transactions/RecentTransactionScreen.kt

```diff
@@ -0,0 +1,275 @@
+                                .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                        )
+                    }
+
+                    is RecentTransactionUiState.LoadMoreRecentTransactions -> {
+                        LazyColumn(
+                            modifier = Modifier.fillMaxSize(),
+                            state = lazyColumnState
+                        ) {
+                            items(uiState.transactions) { transaction ->
+                                RecentTransactionListItem(transaction)
+                            }
+                        }
+                    }
+
+                    is RecentTransactionUiState.RecentTransactions -> {
+
+                        if (uiState.transactions?.isNotEmpty() == true) {
+                            LoadRecentTransactions(transactionList = uiState.transactions as ArrayList<Transaction>,lazyColumnState,uiState, viewModel)
+                        } else {
+                            MifosErrorComponent(isEmptyData = true)
+                        }
+                    }
+                }
+            }
+
+            if (pullRefreshState.isRefreshing) {
+                LaunchedEffect(key1 = true) {
+                    onRefresh()
+                }
+            }
+            LaunchedEffect(key1 = isRefreshing) {
+                if (isRefreshing)
+                    pullRefreshState.startRefresh()
+                else
+                    pullRefreshState.endRefresh()
+            }
+
+            PullToRefreshContainer(
+                state = pullRefreshState,
+                modifier = Modifier.align(Alignment.TopCenter),
+            )
+    }
+
+}
+
+
+@Composable
+fun LoadRecentTransactions(
+    transactionList: ArrayList<Transaction>,
+    lazyColumnState: LazyListState,
+    uiState: RecentTransactionUiState.RecentTransactions,
+    viewModel: RecentTransactionViewModel
+){
+
+    LazyColumn(Modifier.fillMaxSize(),
+        state = lazyColumnState) {
+        items(items = transactionList?.toList().orEmpty()) {
+            RecentTransactionListItem(it)
+        }
+        item {
+                val visibleItems = lazyColumnState.layoutInfo.visibleItemsInfo
+                val lastVisibleItemIndex = visibleItems.lastOrNull()?.index ?: 0
+                val isNearBottom = lastVisibleItemIndex >= uiState.transactions.size - 5
+
+                if (isNearBottom) {
+                    viewModel.loadRecentTransactions(true, uiState.transactions.size)
+                }
+        }
+    }
+
+}
+
+
+@Composable
+fun RecentTransactionListItem(transaction: Transaction?) {
+    Row(modifier = Modifier.padding(8.dp), verticalAlignment = Alignment.CenterVertically) {
+        Image(
+            painter = painterResource(id = R.drawable.ic_local_atm_black_24dp),
+            contentDescription = stringResource(id = R.string.atm_icon),
+            modifier = Modifier.size(39.dp)
```

Comment:
will be better if we can use material guidelines. 39 dp doesn't make much sense .